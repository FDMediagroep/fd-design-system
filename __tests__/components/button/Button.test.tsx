import React from 'react';
import {
    Button,
    getCssClassNames,
} from '../../../src/components/button/Button';
import { render, fireEvent } from '@testing-library/react';

describe('Button', () => {
    test('should render correctly', () => {
        const { getByText } = render(<Button>Knopje</Button>);

        expect(getByText('Knopje')).toBeTruthy();
    });

    test('should handle clicks correctly', () => {
        const spy = jest.fn();
        const { getByText, container } = render(
            <Button onClick={spy}>Knopje</Button>
        );

        expect(getByText('Knopje')).toBeTruthy();
        expect(spy).toBeCalledTimes(0);
        fireEvent.click(container.querySelector('button'));
        expect(spy).toBeCalledTimes(1);
        fireEvent.click(container.querySelector('button'));
        expect(spy).toBeCalledTimes(2);
    });

    test('should handle sizes correctly', () => {
        const { container } = render(<Button size="s">Knopje</Button>);
        const { container: container2 } = render(
            <Button size="m">Knopje</Button>
        );

        expect(container).toMatchSnapshot();
        expect(container2).toMatchSnapshot();
    });

    test('should have correct class name', () => {
        const { container } = render(<Button>Knopje</Button>);

        expect(
            container.firstElementChild?.classList.contains(
                getCssClassNames()[0]
            )
        ).toBeTruthy();
    });
});
