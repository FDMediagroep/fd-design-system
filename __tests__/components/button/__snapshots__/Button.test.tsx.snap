// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Button should handle sizes correctly 1`] = `
<div>
  <button
    class="fdButton "
  >
    Knopje
  </button>
</div>
`;

exports[`Button should handle sizes correctly 2`] = `
<div>
  <button
    class="fdButton m"
  >
    Knopje
  </button>
</div>
`;
